ui/store/actions.ts

```diff
@@ -3695,8 +3695,13 @@ export function resolvePendingApproval(
     await submitRequestToBackground('resolvePendingApproval', [id, value]);
     // Before closing the current window, check if any additional confirmations
     // are added as a result of this confirmation being accepted
-    const { pendingApprovals } = await forceUpdateMetamaskState(dispatch);
-    if (Object.values(pendingApprovals).length === 0) {
+    const { pendingApprovals, approvalFlows } = await forceUpdateMetamaskState(
+      dispatch,
+    );
+    if (
+      Object.values(pendingApprovals).length === 0 &&
+      approvalFlows.length === 0
```

Comment:
Rather than putting this logic here, we could instead add it lower inside `closeCurrentNotificationWindow` as that also has conditions before closing, so it would cover more use cases which call it directly.

I remember having to change this when adding a flow to the transactions for example.

Reply:
I also think it makes sense not to hide the dispatch here and in `rejectPendingApproval` behind an `if` statement, but I'm a bit worried about any unintended side effects that might have.

Reply:
We can address in a future ticket as use cases require, but some paths simply call `closeCurrentNotificationWindow` directly and so we'll want to prevent that if we have approval flows.

Reply:
Done. I have just moved the `approvalFlow` condition though.